src/make-gecko-parameterized-roles.js

```diff
@@ -0,0 +1,85 @@
+import editRole from './util/edit-role';
+import {getProjects, hgmoPath, scmLevel, feature} from './util/projects';
+
+module.exports.setup = (program) => {
+  return program
+    .command('make-gecko-parameterized-roles')
+    .option('-n, --noop', 'Don\'t change roles, just show difference')
+    .description('make the parameterized roles used by other gecko roles');
+};
+
+var description = desc => [
+  '*DO NOT EDIT*',
+  '',
+  desc,
+  '',
+  'This role is configured automatically by [taskcluster-admin](https://github.com/taskcluster/taskcluster-admin).',
+].join('\n');
+
+module.exports.run = async function(options) {
+  var projects = await getProjects();
+
+  var allLevels = new Set(Object.values(projects).map(proj => scmLevel(proj)));
+  var allDomains = new Set(Object.values(projects).map(proj => proj.trust_domain));
+
+  for (let level of allLevels) {
+    for (let domain of allDomains) {
+      await editRole({
+        roleId: `project:releng:branch:${domain}:level-${level}:*`,
+        description: description(
+          `Scopes for ${domain} projects at level ${level}; the '*' matches the project name.`
+        ),
+        scopes: [
+          `assume:moz-tree:level:${level}:${domain}`,
+          `queue:route:index.${domain}.v2.<..>.*`,
+          `index:insert-task:${domain}.v2.<..>.*`,
+          `queue:route:index.${domain}.cache.level-${level}.*`,
+          `index:insert-task:${domain}.cache.level-${level}.*`,
+          `queue:route:tc-treeherder-stage.<..>.*`,
+          `queue:route:tc-treeherder.<..>.*`,
+          `queue:route:tc-treeherder-stage.v2.<..>.*`,
+          `queue:route:tc-treeherder.v2.<..>.*`,
+          `queue:route:coalesce.v1.builds.<..>.*`,  // deprecated - https://bugzilla.mozilla.org/show_bug.cgi?id=1382204
```

Comment:
This and the tc-treeherder-stage routes are old, and could perhaps be moved to a deprecated feature? Although, I guess maybe that wants to be done separately.